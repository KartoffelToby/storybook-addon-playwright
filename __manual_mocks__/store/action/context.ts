import { ReducerState } from '../../../src/store/actions/reducer';
import { useActionDispatchContext } from '../../../src/store/actions/ActionContext';
import { mocked } from 'ts-jest/utils';

jest.mock('../../../src/store/actions/ActionContext');

export const dispatchMock = jest.fn();

const useActionDispatchContextMock = mocked(useActionDispatchContext);

useActionDispatchContextMock.mockImplementation(() => {
  return (...arg) => {
    return dispatchMock(arg);
  };
});

export { ReducerState };
