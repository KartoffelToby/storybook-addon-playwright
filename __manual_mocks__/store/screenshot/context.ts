import { ReducerState } from '../../../src/store/screenshot/reducer';

export const dispatchMock = jest.fn();

const mockData: Partial<ReducerState> = {
  screenshots: [],
};

export const useScreenShotContext = jest.fn() as jest.Mock<
  Partial<ReducerState>
>;
useScreenShotContext.mockImplementation(() => mockData);

jest.mock('../../../src/store/screenshot/context', () => ({
  useScreenshotContext: useScreenShotContext,
  useScreenshotDispatch: () => {
    return (...arg) => {
      return dispatchMock(arg);
    };
  },
}));

export { ReducerState };

export default ReducerState;
