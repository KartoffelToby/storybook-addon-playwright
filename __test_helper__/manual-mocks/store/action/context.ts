import { getActionSchemaData } from '../../../action-schema';
import { ReducerState } from '../../../../src/store/actions/reducer';

export const dispatchMock = jest.fn();

const mockData: Partial<ReducerState> = {
  actionSchema: getActionSchemaData(),
};

export const useActionContext = jest.fn() as jest.Mock<Partial<ReducerState>>;
useActionContext.mockImplementation(() => mockData);

jest.mock('../../../../src/store/actions/ActionContext', () => ({
  useActionContext,
  useActionDispatchContext: () => {
    return (...arg) => {
      return dispatchMock(arg);
    };
  },
}));

export { ReducerState };
