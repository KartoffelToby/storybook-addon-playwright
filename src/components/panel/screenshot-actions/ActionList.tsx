import React, { SFC, useMemo } from 'react';
import { ActionOptions } from './ActionOptions';
import { useActionSetActions } from '../../../hooks';
import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';
import DragIndicatorSharp from '@material-ui/icons/DragIndicatorSharp';

const _dragEl = document.getElementById('draggable');

const getListStyle = (isDraggingOver) => ({
  // position: 'static',
});

const getItemStyle = (isDragging, draggableStyle) => ({
  // some basic styles to make the items look a bit nicer

  // change background colour if dragging
  userSelect: 'none',

  // styles we need to apply on draggables
  ...draggableStyle,
});

const ActionList: SFC = () => {
  const { actionSetActions } = useActionSetActions();

  // return useMemo(() => {
  if (!actionSetActions.length) return null;
  return (
    <DragDropContext onDragEnd={this.onDragEnd}>
      <Droppable droppableId="droppable">
        {(droppableProvided, droppableSnapshot) => (
          <div
            ref={droppableProvided.innerRef}
            style={getListStyle(droppableSnapshot.isDraggingOver)}
          >
            {actionSetActions.map((action, index) => (
              <Draggable key={action.id} draggableId={action.id} index={index}>
                {(draggableProvided, draggableSnapshot) => (
                  <div
                    ref={draggableProvided.innerRef}
                    {...draggableProvided.draggableProps}
                    style={getItemStyle(
                      draggableSnapshot.isDragging,
                      draggableProvided.draggableProps.style,
                    )}
                  >
                    <ActionOptions
                      key={action.id}
                      actionName={action.name}
                      actionId={action.id}
                      DragHandle={() => (
                        <div {...draggableProvided.dragHandleProps}>
                          <DragIndicatorSharp />
                        </div>
                      )}
                    />
                  </div>
                )}
              </Draggable>
            ))}
            {droppableProvided.placeholder}
          </div>
        )}
      </Droppable>
    </DragDropContext>
  );
  // }, [actionSetActions]);
};

ActionList.displayName = 'ActionList';

export { ActionList };
