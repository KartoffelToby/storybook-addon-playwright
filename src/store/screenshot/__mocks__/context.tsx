import { ReducerState } from '../reducer';

export const ScreenshotDispatchContext = jest.fn();

const useScreenshotDispatch = jest.fn();
useScreenshotDispatch.mockImplementation(() => {
  return jest.fn();
});

const useScreenshotContext = jest.fn();

const StateData: Partial<ReducerState> = {
  imageDiffResults: [{ pass: true, screenshotHash: 'hash' }],
  screenshots: [
    { browserType: 'chromium', hash: 'hash', index: 0, title: 'title' },
    { browserType: 'chromium', hash: 'hash-2', index: 0, title: 'title' },
  ],
};

useScreenshotContext.mockImplementation((): Partial<ReducerState> => StateData);

export { useScreenshotDispatch, useScreenshotContext };
