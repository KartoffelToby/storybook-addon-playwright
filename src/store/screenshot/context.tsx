import React, { createContext, useContext, useEffect } from 'react';
import { initialState, reducer, ReducerState, Action } from './reducer';
import { useReducer } from 'reinspect';
import {
  useGlobalScreenshotDispatch,
  useGlobalImageDiffResults,
} from '../../hooks';

export const ScreenshotDispatchContext = React.createContext<
  React.Dispatch<Action>
>({} as React.Dispatch<Action>);

export const useScreenshotDispatch = () =>
  useContext(ScreenshotDispatchContext);

export const ScreenshotContext = createContext<ReducerState>(
  {} as ReducerState,
);
export const useScreenshotContext = () => useContext(ScreenshotContext);

export const ScreenshotContextProvider = ScreenshotContext.Provider;
export const ScreenshotConsumer = ScreenshotContext.Consumer;

const ScreenshotProvider: React.FC = (props) => {
  const { children } = props;

  const [state, dispatch] = useReducer(
    reducer,
    initialState,
    () => initialState,
    'Screenshot',
  );

  useGlobalScreenshotDispatch((action) => {
    dispatch(action);
  });

  const { setImageDiffResult } = useGlobalImageDiffResults();

  useEffect(() => {
    setImageDiffResult(state.imageDiffResults.filter((x) => !x.pass));
  }, [setImageDiffResult, state.imageDiffResults]);

  return (
    <ScreenshotContextProvider value={state}>
      <ScreenshotDispatchContext.Provider value={dispatch}>
        {children}
      </ScreenshotDispatchContext.Provider>
    </ScreenshotContextProvider>
  );
};

export { ScreenshotProvider };
