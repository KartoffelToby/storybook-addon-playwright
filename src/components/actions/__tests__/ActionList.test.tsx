import { dispatchMock } from '../../../../__manual_mocks__/store/action/context';
import React from 'react';
import { ActionList, SortableList, SortableItem } from '../ActionList';
import { shallow } from 'enzyme';
import { SortEnd, SortEvent } from 'react-sortable-hoc';

jest.mock('../../../hooks/use-current-story-data.ts');

describe('ActionList', () => {
  beforeEach(() => {
    dispatchMock.mockClear();
  });

  it('should mount', () => {
    const wrapper = shallow(
      <ActionList
        actionSet={{
          actions: [{ id: 'action-id', name: 'action-name' }],
          id: 'action-set-id',
          title: 'action-set-desc',
        }}
      />,
    );
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.find(SortableList).exists()).toBeTruthy();
  });

  it('should show error if no actions provided', () => {
    const wrapper = shallow(
      <ActionList
        actionSet={{
          actions: [],
          id: 'action-set-id',
          title: 'action-set-desc',
        }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle sort end', () => {
    const wrapper = shallow(
      <ActionList
        actionSet={{
          actions: [{ id: 'action-id', name: 'action-name' }],
          id: 'action-set-id',
          title: 'action-set-desc',
        }}
      />,
    );
    const sortableList = wrapper.find(SortableList);

    sortableList
      .props()
      .onSortEnd({ newIndex: 1, oldIndex: 2 } as SortEnd, {} as SortEvent);

    expect(dispatchMock).toHaveBeenCalledTimes(1);
  });

  it('should have items', () => {
    const wrapper = shallow(
      <SortableList
        items={[
          { id: 'action-id-1', name: 'action-name-1' },
          { id: 'action-id-2', name: 'action-name-2' },
        ]}
        actionSetId="action-set-id"
      />,
      { disableLifecycleMethods: true },
    )
      .first()
      .shallow();

    const items = wrapper.find(SortableItem);

    expect(items).toHaveLength(2);
  });
});
