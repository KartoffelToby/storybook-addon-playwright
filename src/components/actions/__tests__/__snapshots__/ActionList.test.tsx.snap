// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionList should show error id no actions provided 1`] = `
<div
  className="ActionSetList-noActionMessage-1"
>
  <div>
    No action to display!
  </div>
  <div>
    Click the 
    '+'
     button to create an action.
  </div>
</div>
`;
